### PR TITLE
Chore: Add context to star and stats

### PR DESCRIPTION
**Pull request by idafurjes**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Ref #36734

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/sqlstore/star.go [hunk 1]
```diff
@@ -49,19 +49,19 @@ func StarDashboard(cmd *models.StarDashboardCommand) error {
...
-	return inTransaction(func(sess *DBSession) error {
+	return ss.WithTransactionalDbSession(ctx, func(sess *DBSession) error {
 		var rawSQL = "DELETE FROM star WHERE user_id=? and dashboard_id=?"
 		_, err := sess.Exec(rawSQL, cmd.UserId, cmd.DashboardId)
 		return err
 	})
 }
 
-func GetUserStars(query *models.GetUserStarsQuery) error {
+func (ss *SQLStore) GetUserStars(ctx context.Context, query *models.GetUserStarsQuery) error {
```
Comment (marefr) [inline pkg/services/sqlstore/star.go:64]: Please use `ss.WithDbSession` below instead of `x`.

File: pkg/services/sqlstore/star.go [hunk 2]
```diff
@@ -7,17 +7,17 @@ import (
...
+	bus.AddHandlerCtx("sql", ss.StarDashboard)
+	bus.AddHandlerCtx("sql", ss.UnstarDashboard)
+	bus.AddHandlerCtx("sql", ss.GetUserStars)
+	bus.AddHandlerCtx("sql", ss.IsStarredByUserCtx)
 }
 
-func IsStarredByUserCtx(ctx context.Context, query *models.IsStarredByUserQuery) error {
-	return withDbSession(ctx, x, func(dbSession *DBSession) error {
+func (ss *SQLStore) IsStarredByUserCtx(ctx context.Context, query *models.IsStarredByUserQuery) error {
+	return ss.WithTransactionalDbSession(ctx, func(sess *DBSession) error {
```
Comment (marefr) [inline pkg/services/sqlstore/star.go:18]: Query only reads so no need for a transaction.

```suggestion
	return ss.WithDbSession(ctx, func(sess *DBSession) error {
```

**Review by marefr — CHANGES_REQUESTED** [review]
Looks good. Had been easier to review by separating stats and star changes in two separate PRs, but my own personal preference. Stats-changes feels inconsistent compared with stars (using sqlstore vs global `x` variable), but maybe that causing too many changes? Planning to fix that later?

Think you might want to update this as well to use context:
https://github.com/grafana/grafana/blob/58d69d10555339d2363155b8cfd560ea79f18859/pkg/api/admin.go#L25

**Comment by idafurjes** [comment]
I felt the PR was a bit small, that's why I added the star

**Comment by marefr** [comment]
> I felt the PR was a bit small, that's why I added the star

I guess everything is relative :smile: But in general, keeping it small could allow you to get a PR reviewed/merged faster and by that maybe feeling more productive, mark that as done and move on to a different task rather than having multiple semi-small/larger PRs going on in parallel. 

Let me know when this is ready for review - still a draft.

**Review by marefr — APPROVED** [review]
LGTM, but build seems to have failed. I restarted it

**Pull request merged**